data-prepper-plugins/mongodb/src/main/java/org/opensearch/dataprepper/plugins/mongo/stream/StreamWorker.java

```diff
@@ -72,6 +77,20 @@ public class StreamWorker {
     private final JsonWriterSettings writerSettings = JsonWriterSettings.builder()
             .outputMode(JsonMode.RELAXED)
             .objectIdConverter((value, writer) -> writer.writeString(value.toHexString()))
+            .binaryConverter((value, writer) ->  writer.writeString(Base64.getEncoder().encodeToString(value.getData())))
+            .dateTimeConverter((value, writer) -> writer.writeNumber(String.valueOf(value.longValue())))
+            .decimal128Converter((value, writer) -> writer.writeString(value.bigDecimalValue().toPlainString()))
+            .doubleConverter((value, writer) -> writer.writeNumber(String.valueOf(value.doubleValue())))
```

Comment:
Why convert the double value to a string first?

Would this work?

```suggestion
            .doubleConverter((value, writer) -> writer.writeNumber(value.doubleValue()))
```

Reply:
We can use default relaxed double conversion. I will remove this.